### PR TITLE
speed up 128-bit fft using faster sum, mul, and using more lanes with avx512

### PR DESCRIPTION
**Pull request by sarah-ek**

### CODE REVIEW

File: src/fft128/f128_ops.rs [hunk 1]
```diff
@@ -705,6 +687,67 @@ pub mod x86 {
         (p, simd.mul_sub_f64x8(a, b, p))
     }
 
+    #[cfg(feature = "nightly")]
+    #[inline(always)]
+    pub(crate) fn quick_two_sum_f64x16(simd: V4, a: f64x16, b: f64x16) -> (f64x16, f64x16) {
+        let s = simd.add_f64x16(a, b);
+        (s, simd.sub_f64x16(b, simd.sub_f64x16(s, a)))
```
Comment (IceTDrinker) [inline src/fft128/f128_ops.rs:694]: this looks different than the x8 and x4 variants ? Why is that so ?

File: src/fft128/mod.rs [hunk 2]
```diff
@@ -2112,7 +2306,7 @@ mod tests {
     #[test]
     fn test_product_avx512() {
         if let Some(simd) = V4::try_new() {
```
Comment (IceTDrinker) [inline src/fft128/mod.rs:2308]: don't we need additional tests for V4x2 ?

File: src/fft128/f128_ops.rs (hunk 1 above)
Comment (sarah-ek) [reply src/fft128/f128_ops.rs:694]: what's the difference?

File: src/fft128/mod.rs (hunk 2 above)
Comment (sarah-ek) [reply src/fft128/mod.rs:2308]: both `V4` and `V4x2` are used in the current implementation

File: src/fft128/f128_ops.rs (hunk 1 above)
Comment (IceTDrinker) [reply src/fft128/f128_ops.rs:694]: ah no it's my bad, I misread, all good

File: src/fft128/mod.rs (hunk 2 above)
Comment (IceTDrinker) [reply src/fft128/mod.rs:2308]: ok perfect !

**Review by IceTDrinker — APPROVED** [review]
Good to merge ! Do you want to prepare a patch release with this improvement or should we wait for the next work you want to do on f128 ?

**Comment by sarah-ek** [comment]
i think we can do a release for now. the recursive impl stuff is gonna take a bit of time to finish and it's backward compatible so we don't need to worry about a minor version bump

**Review by IceTDrinker — APPROVED** [review]

**Comment by IceTDrinker** [comment]
> done!

Thanks a lot, we'll wait for the CI and then we can publish the release, we'll update the concrete-fft version in TFHE-rs to benefit from the speed up :)

**Pull request merged**